src/platform/site-wide/user-nav/actions/index.js

```diff
@@ -9,7 +11,34 @@ export function toggleSearchHelpUserMenu(menu, isOpen) {
   };
 }
 
-export function toggleLoginModal(isOpen) {
+function logGAEvent(context) {
+  const toolContexts = [
+    'rx',
+    'messaging',
+    'appointments',
+    'lab-and-test-results',
+    'health-records',
+    'letters',
+  ];
+
+  let eventContext;
+
+  // If the context provided is in the list above, set context as "tools", otherwise
+  // use the context provided.  If no context, default to "forms".
+  if (context) {
+    eventContext = toolContexts.indexOf(context) !== -1 ? 'tools' : context;
+  } else {
+    eventContext = 'forms';
+  }
+
+  recordEvent({ event: `login-link-clicked-${eventContext}` });
+}
+
+export function toggleLoginModal(isOpen, context) {
+  if (isOpen) {
+    logGAEvent(context);
```

Comment:
I think `toggleLoginModal` can be triggered from places other than the listed tools or forms, or even without clicking a link as well, which could mean that the events are inaccurate. The edge cases exist particularly in multiple places in this file:

https://github.com/department-of-veterans-affairs/vets-website/blob/8afbcdd1a09ed4047156a6e332230d0afd0208c6/src/platform/site-wide/user-nav/containers/Main.jsx#L73

https://github.com/department-of-veterans-affairs/vets-website/blob/8afbcdd1a09ed4047156a6e332230d0afd0208c6/src/platform/site-wide/user-nav/containers/Main.jsx#L109

A user can get redirected to the homepage with a `next` query param because an app is gated by login, and 'forms' would be an inaccurate context in that case.

We could try to account for a different context here. Or we could be more explicit about the login link being clicked by doing `recordEvent` in direct response to the click events for specific links. For forms, I believe you would only need to make one change in `SignInLink`, and for the tools, you would only need to make the change in the CTA component.

Reply:
Good point.  I've moved `recordEvent` to the `CTAWidget`, `SignInLink` and `SearchHelpSignIn`.  For `CTAWidget` I am still filtering those specific apps to use the `tools` context, as I believe the `CTAWidget` is also used in other contexts.